### PR TITLE
Hyperparameter Tuning and Training Run Comparison

### PR DESCRIPTION
**Pull request by shivv23**
## Overview

This PR adds hyperparameter tuning and training run comparison capabilities to Tensormap, helping users optimize their models and track training history.

### Hyperparameter Tuning
Grid search over key hyperparameters to find optimal configuration:
- **Learning Rate**: [0.001, 0.01]
- **Batch Size**: [16, 32]
- **Epochs**: [5, 10]

Returns top 5 configurations sorted by loss, with best configuration highlighted.

### Training Run Comparison
Track and compare metrics across multiple training runs:
- Chronological view of runs with loss, accuracy, epochs
- Automatic best run identification (lowest loss)
- Summary statistics for project

## Technical Details

**Hyperparameter Tuning Endpoint:** `GET /model/tune/{model_name}`

**Run Comparison Endpoint:** `GET /model/compare`

**Query Parameters:**
- `project_id` (optional): Filter by project
- `file_id` (optional): Use specific dataset
- `limit` (optional, default: 10): Number of runs to return

**Response (Tuning):**
```json
{
  "success": true,
  "message": "Tested 8 configurations",
  "data": {
    "results": [
      {"learning_rate": 0.001, "batch_size": 32, "epochs": 10, "loss": 0.1245}
    ],
    "best": {"learning_rate": 0.001, "batch_size": 32, "epochs": 10, "loss": 0.1245},
    "total_tested": 8
  }
}
```

### Files Changed
- `app/services/deep_learning.py`: Added `tune_hyperparameters_service()` and `compare_runs_service()`
- `app/routers/deep_learning.py`: Added endpoints
- `tests/test_tune.py`: Unit tests

## GSoC Alignment

This implements:
- **Blueprint #6: Hyperparameter Tuning**
- **Blueprint #4: Run Comparison**

### CODE REVIEW

**Comment by shivv23** [comment]
Merged features via main branch
